magefiles/installation/install.go

```diff
@@ -149,6 +164,149 @@ func (i *InstallAppStudio) cloneInfraDeployments() (*git.Remote, error) {
 	return repo.CreateRemote(&config.RemoteConfig{Name: i.LocalForkName, URLs: []string{fmt.Sprintf("https://github.com/%s/infra-deployments.git", i.LocalGithubForkOrganization)}})
 }
 
+func MergePRInRemote(branch string, forkOrganization string, repoPath string) error {
+	if branch == "" {
+		klog.Fatal("The branch for upgrade is empty!")
+	}
+
+	repo, err := git.PlainOpen(repoPath)
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	branches, err := repo.Branches()
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	var branchRepo string
+	err = branches.ForEach(func(ref *plumbing.Reference) error {
+		if !strings.Contains("main", ref.Name().String()) {
+			branchRepo = ref.Name().String()
+		}
+		return nil
+	})
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	wt, err := repo.Worktree()
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	err = wt.Checkout(&git.CheckoutOptions{
+		Branch: plumbing.ReferenceName(branchRepo),
+	})
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	if forkOrganization == "" {
+		_, err = exec.Command("git", "-C", repoPath, "merge", "remotes/origin/"+branch, "-q").Output()
```

Comment:
I guess you did this because git-go doesn't support merges, rigth?
Looking at https://pkg.go.dev/github.com/go-git/go-git/v5#Worktree.Pull it could support merge if it can be resolved fast-forward. This should be allways our situation, no? Have you tried this? Can you try this?

Reply:
We cannot use the go-git pull - it supports only fast forward merge, but we need to use merge - because of the preview commit and some changes in infra PRs could introduce conflicts and right now we are solving it with "theirs" merge strategy.